conda/gateways/connection/download.py

```diff
@@ -177,14 +178,17 @@ def check(target):
         if md5 or sha256:
             checksum_type = "sha256" if sha256 else "md5"
             checksum = sha256 if sha256 else md5
+            try:
+                checksum_bytes = bytes.fromhex(checksum)
+            except (ValueError, TypeError) as exc:
+                raise CondaValueError(exc) from exc
```

Comment:
Isn't the `from exc` already the default behavior?

Reply:
What do you mean by default behavior here? The output is different in this test:

```
>>> try:
...   raise ValueError("First exception")
... except ValueError as exc:
...   raise TypeError(exc) from exc
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ValueError: First exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
TypeError: First exception
>>> try:
...   raise ValueError("First exception")
... except ValueError as exc:
...   raise TypeError(exc)
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ValueError: First exception

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
TypeError: First exception
```

Reply:
Yea, I wouldn't force this on every contributor. I do it out of habit when I'm explicitly turning one exception into another so there's a clear intent for that papertrail. But the difference might be too small to make it, e.g., an imposed code style rule for the project. There's a nice discussion about this in [this SO answer](https://stackoverflow.com/questions/24752395/python-raise-from-usage) and in the [originating PEP](https://legacy.python.org/dev/peps/pep-3134/).